test/router-luatest/reload_test.lua

```diff
@@ -0,0 +1,265 @@
+    vtest.drop_instance(g, router)
+end
+
+local function test_master_search_template(g, router, auto_master_cfg)
+    vtest.router_cfg(router, auto_master_cfg)
+
+    -- Working with first replicaset (2 instances)
+    local rs_uuid, master_uuid = unpack(g.replica_1_a:exec(function()
+        local info = box.info()
+        return {info.cluster.uuid, info.uuid}
+    end))
+
+    local replica_uuid = g.replica_1_b:exec(function()
+        return box.info.uuid
+    end)
+
+    local function stop_master_search()
+        ivshard.router.internal.errinj.ERRINJ_MASTER_SEARCH_DELAY = true
+        ilt.helpers.retrying({timeout = ivtest.wait_timeout}, function()
+            local errinj = ivshard.router.internal.errinj
+            if errinj.ERRINJ_MASTER_SEARCH_DELAY ~= 'in' then
+                error('Master search is not stopped yet')
+            end
+        end)
+    end
+
+    router:exec(function()
+        local router = ivshard.router.internal.static_router
+        ivtest.service_wait_for_ok(router.master_search_service,
+            {on_yield = ivshard.router.master_search_wakeup})
+        for _, rs in pairs(router.replicasets) do
+            ilt.assert_not_equals(rs.master, nil)
+        end
+    end)
+
+    router:exec(stop_master_search)
+
+    -- Change master and wait, until it's found
+    global_cfg.sharding[rs_uuid].replicas[master_uuid].master = false
+    global_cfg.sharding[rs_uuid].replicas[replica_uuid].master = true
+    vtest.cluster_cfg(g, global_cfg)
+
+    local function continue_and_check_master_search(rs, master)
+        ivshard.router.internal.errinj.ERRINJ_MASTER_SEARCH_DELAY = false
+        local router = ivshard.router.internal.static_router
+        ivtest.service_wait_for_new_ok(router.master_search_service,
+            {on_yield = ivshard.router.master_search_wakeup})
+        ilt.assert_equals(master, router.replicasets[rs].master.uuid)
+    end
+
+    router:exec(continue_and_check_master_search, {rs_uuid, replica_uuid})
+    router:exec(stop_master_search)
+
+    -- Restore configuration of the cluster
+    global_cfg.sharding[rs_uuid].replicas[master_uuid].master = true
+    global_cfg.sharding[rs_uuid].replicas[replica_uuid].master = false
+    vtest.cluster_cfg(g, global_cfg)
+
+    router:exec(continue_and_check_master_search, {rs_uuid, master_uuid})
+end
+
+g.test_master_search = function(g)
+    local hash = 'f5f386a5a35e6e5efd8f4f2ed1b3d208fdae9095'
+    local router = create_router_at(hash)
+
+    -- Enable auto master search
+    local auto_master_cfg_template = table.deepcopy(cfg_template)
+    for _, rs in pairs(auto_master_cfg_template.sharding) do
+        rs.master = 'auto'
+        for _, r in pairs(rs.replicas) do
+            r.master = nil
+        end
+    end
+    local auto_master_cfg = vtest.config_new(auto_master_cfg_template)
+
+    -- Service_info was introduced in the following commit:
+    --     "router: add saving of background service statuses".
```

Comment:
Why mention it if this test case is about master search?

Reply:
Because the test uses it, so old version must have service_info. I extended comment